File: CPP/Program/Vehicle.cpp
//menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

//memasukkan library yang digunakan
#include <iostream>
#include <string>

//deklarasi namespace
using namespace std;

//kelas Vehicle yang digunakan untuk merepresentasikan data Vehicle, Vehicle merupakan kelas induk dari kelas Car dan Motorcycle
class Vehicle{
    private:
        //atribut yang digunakan pada kelas Vehicle terdiri dari id, plat nomor, merk, tahun produksi, dan warna
        string kode;
        string plat_nomor;
        string merk;
        int tahun_produksi;
        string warna;

    public:
        //constructor
        Vehicle(){

        }

        //constructor: overloading
        Vehicle(string kode, string plat_nomor, string merk, int tahun_produksi, string warna){
            this->kode = kode;
            this->plat_nomor = plat_nomor;
            this->merk = merk;
            this->tahun_produksi = tahun_produksi;
            this->warna = warna;
        }

        //menampilkan atribut vehicle
        virtual void displayInfo() const{
            cout << "Kode Kendaraan     : " << kode << endl;
            cout << "Plat Nomor         : " << plat_nomor << endl;
            cout << "Merk               : " << merk << endl;
            cout << "Tahun Produksi     : " << tahun_produksi << endl;
            cout << "Warna              : " << warna << endl;
        }

        //method setter untuk set nilai setiap atribut pada kelas Vehicle
        //set kode
        void setKode(string kode){
            this->kode = kode;
        }

        //set plat nomor
        void setPlatNomor(string plat_nomor){
            this->plat_nomor = plat_nomor;
        }

        //set merk
        void setMerk(string merk){
            this->merk = merk;
        }

        //set tahun produksi
        void setTahunProduksi(int tahun_produksi){
            this->tahun_produksi = tahun_produksi;
        }

        //set warna
        void setWarna(string warna){
            this->warna = warna;
        }

        //method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
        //get kode
        string getKode(){
            return kode;
        }

        //get plat nomor
        string getPlatNomor(){
            return plat_nomor;
        }

        //get merk
        string getMerk(){
            return merk;
        }

        //get tahun produksi
        int getTahunProduksi(){
            return tahun_produksi;
        }

        //get warna
        string getWarna(){
            return warna;
        }

        //destructor
        virtual ~Vehicle() {

        }
};

File: Python/Program/Vehicle.py
# deklarasi class
class Vehicle:
    # constructor
    def __init__(self, kode_kendaraan, plat_nomor, merk, tahun_produksi, warna):
        self.kode_kendaraan = kode_kendaraan
        self.plat_nomor = plat_nomor
        self.merk = merk
        self.tahun_produksi = tahun_produksi
        self.warna = warna

    # menampilkan atribut vehicle
    def display_info(self):
        print("Kode Kendaraan     :", self.kode_kendaraan)
        print("Plat Nomor         :", self.plat_nomor)
        print("Merk               :", self.merk)
        print("Tahun Produksi     :", self.tahun_produksi)
        print("Warna              :", self.warna)
    
    # method setter untuk set nilai setiap atribut pada kelas Vehicle
    # set kode kendaraan
    def set_kode_kendaraan(self, kode_kendaraan):
        self.kode_kendaraan = kode_kendaraan
    
    # set plat_nomor
    def set_plat_nomor(self, plat_nomor):
        self.plat_nomor = plat_nomor
        
    # set merk
    def set_merk(self, merk):
        self.merk = merk
        
    # set tahun produksi
    def set_tahun_produksi(self, tahun_produksi):
        self.tahun_produksi = tahun_produksi
        
    # set warna
    def set_warna(self, warna):
        self.warna = warna
        
    # method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
    # get kode_kendaraan
    def get_kode_kendaraan(self):
        return self.kode_kendaraan

    # get plat nomor
    def get_plat_nomor(self):
        return self.plat_nomor
    
    # get merk
    def get_merk(self):
        return self.merk
    
    # get tahun produksi
    def get_tahun_produksi(self):
        return self.tahun_produksi
    
    # get warna
    def get_warna(self):
        return self.warna